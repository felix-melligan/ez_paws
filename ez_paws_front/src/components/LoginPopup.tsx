import React, { Component } from 'react';
import {
    Modal,
    Button,
    Form
} from 'react-bootstrap';
import { Link } from 'react-router-dom';

type MyProps = {
    show: boolean,
    onHide: () => void,
};

type MyState = {
    email: string,
    password: string
};

class LoginPopup extends Component<MyProps, MyState> {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            password: ''
        };

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleFormChange = this.handleFormChange.bind(this);
    };

    handleSubmit(): void {
        this.props.onHide();
    };

    handleFormChange(event: any): void {
        const state: string = event.target.type;
        const updatedValue: string = event.target.value;
        this.setState({
            [state]: updatedValue
        } as any);
    };

    render() {
        return (
            <Modal
              {...this.props}
              size="lg"
              aria-labelledby="contained-modal-title-vcenter"
              centered
            >
              <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-vcenter">
                  Login / SignUp
                </Modal.Title>
              </Modal.Header>
              <Modal.Body>
                <p>
                    Existing users: Please use the form to log in.
                </p>
                <Form>
                    <Form.Group controlId="formBasicEmail">
                        <Form.Label>Email address</Form.Label>
                        <Form.Control
                            className='SignInFormEmail'
                            value={this.state.email}
                            onChange={this.handleFormChange}
                            type="email"
                            placeholder="Enter email"
                        />
                        <Form.Text className="text-muted">
                            We'll never share your email with anyone else.
                        </Form.Text>
                    </Form.Group>
                    <Form.Group controlId="formBasicPassword">
                        <Form.Label>Password</Form.Label>
                        <Form.Control
                            className='SignInFormPassword'
                            value={this.state.password}
                            onChange={this.handleFormChange}
                            type="password"
                            placeholder="Password"
                        />
                    </Form.Group>
                </Form>
                <p>
                    New users: Please click the "SignUp" link below to create an account.
                </p>
              </Modal.Body>
              <Modal.Footer>
                <Button className='LoginPopupSignUpButton' as={Link} to='/SignUp' onClick={this.props.onHide}>SignUp</Button>
                <Button className='LoginPopupCloseButton' variant="primary" type="submit" onClick={this.handleSubmit}>Submit</Button>
              </Modal.Footer>
            </Modal>
        );
    };
}

export default LoginPopup;