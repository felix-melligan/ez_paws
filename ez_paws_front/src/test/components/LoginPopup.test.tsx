import React from 'react';
import { 
  shallow,
} from 'enzyme';

import LoginPopup from '../../components/LoginPopup';

let wrapper;

const props = {
    show: false,
    onHide: () => {},
};

beforeEach(() => {
    wrapper = shallow(<LoginPopup {...props}/>);
});

describe('LoginPopup', () => {
    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('Handlers work correctly', () => {
        it('Changes the state of password when value is typed into form', () => {
            const password = 'password';

            expect(wrapper.state().password).toEqual('');

            const form = wrapper.find('.SignInFormPassword');
            form.simulate('change', { target: { type: password, value: password } })

            expect(wrapper.state().password).toEqual(password);
        });

        it('Changes the state of email when value is typed into form', () => {
            const email = 'email';

            expect(wrapper.state().email).toEqual('');

            wrapper.find('.SignInFormEmail').simulate('change', { target: { type: email, value: email }});
            
            expect(wrapper.state().email).toEqual(email);
        });
    });
});