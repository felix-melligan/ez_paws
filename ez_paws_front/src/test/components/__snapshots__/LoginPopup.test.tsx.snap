// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPopup Renders correctly 1`] = `ShallowWrapper {}`;
